src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5264,14 +5264,21 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, DiagnosticBa
 
                 if (searchExtensionMethodsIfNecessary)
                 {
-                    return new BoundMethodGroup(
+                    var boundMethodGroup = new BoundMethodGroup(
                         node,
                         typeArguments,
                         boundLeft,
                         rightName,
                         lookupResult.Symbols.All(s => s.Kind == SymbolKind.Method) ? lookupResult.Symbols.SelectAsArray(s_toMethodSymbolFunc) : ImmutableArray<MethodSymbol>.Empty,
                         lookupResult,
                         flags);
+
+                    if (!boundMethodGroup.HasErrors && boundMethodGroup.ResultKind == LookupResultKind.Empty && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
+                    {
+                        Error(diagnostics, ErrorCode.ERR_BadArity, node, rightName, MessageID.IDS_MethodGroup.Localize(), typeArgumentsSyntax.Count);
```

Comment:
>typeArgumentsSyntax.Count [](start = 122, length = 25)

It feels like this error could be misleading. It uses count from syntax, which, in case of ```<>```, is always 1, regardless what the actual arity of the target method. We should test scenario with a method that has two and more type parameters.